trace/trace.go

```diff
@@ -192,8 +192,10 @@ func startSpanInternal(name string, hasParent bool, parent SpanContext, remotePa
 
 	if !hasParent {
 		span.spanContext.TraceID = cfg.IDGenerator.NewTraceID()
+		copy(span.spanContext.SpanID[:], span.spanContext.TraceID[8:])
```

Comment:
This is too surprising a behavior to implementors of IDGenerator. The reason IDGenerator was added, was to allow custom SpanID formats (as used by X-Ray). They need to set the root span ID as well.